File: PointOfSaleTerminal/C++/Logger.cpp
#include <iostream>

#include "Logger.h"

Logger::Logger( std::string loggerName ) : _name( loggerName )
{

}

Logger::~Logger()
{
}

void Logger::Log( std::string message )
{
#ifdef LOGGING_ENABLED
    std::cout << "Log : " << _name << " : " << message << std::endl;
#endif
}

File: PointOfSaleTerminal/Java/src/com/sandbox/Logger.java
package com.sandbox;

public class Logger
{
    private String loggerName;
    
    public Logger( String loggerNameIn )
    {
        loggerName = loggerNameIn;
    }
    
    public void Log( String message )
    {
        System.out.println( "Log: " + loggerName + " :" + message );
    }
}
